R/json.r

```diff
@@ -0,0 +1,83 @@
+to_json <- function(obj) {
+  obj2 <- recursively_enlist_if_named(obj)
+  as.character(jsonlite::toJSON(obj2, auto_unbox = TRUE))
+}
+
+from_json <- function(obj) {
+  simplify_homogeneous_lists(jsonlite::fromJSON(obj, simplifyVector = FALSE))
+}
+
+has_names <- function(obj) {
+  !is.null(names(obj)) && !is.null(Find(function(s) !is.na(s) && (s != ""), names(obj)))
+}
+
+tricky_identity <- function(obj, type) {
+  if (is.null(obj)) { as(NA, type) } else { as(obj, type) }
+}
+
+enlist_if_named <- function(obj) {
+  if(is.atomic(obj) && has_names(obj)) as.list(obj) else obj
+}
+
+recursively_enlist_if_named <- function(obj) {
+  if (is.atomic(obj) || length(obj) == 0) enlist_if_named(obj)
+  else {
+    if (is.data.frame(obj)) {
+      as.data.frame(lapply(obj, recursively_enlist_if_named))
```

Comment:
This is actually code [that michaelochurch wrote back in February](https://github.com/robertzk/microserver/commit/8b221d1c004c26c2200377dc4c50f9d083697a5d).  I don't know why the data frame logic is being used.